core/src/commonMain/kotlin/com/ustadmobile/core/controller/ClazzWorkDetailOverviewPresenter.kt

```diff
@@ -0,0 +1,207 @@
+                    DoorMutableLiveData(questionsAndOptionsWithResponseList)
+        }
+
+        val publicComments = withTimeoutOrNull(2000){
+            db.commentsDao.findPublicByEntityTypeAndUidLive(ClazzWork.CLAZZ_WORK_TABLE_ID,
+                    clazzWorkWithSubmission.clazzWorkUid)
+        }
+        view.clazzWorkPublicComments = publicComments
+
+
+        if(clazzWorkWithSubmission.clazzWorkCommentsEnabled && view.isStudent) {
+            val privateComments = withTimeoutOrNull(2000) {
+                db.commentsDao.findPrivateByEntityTypeAndUidAndForPersonLive2(ClazzWork.CLAZZ_WORK_TABLE_ID,
+                        clazzWorkWithSubmission.clazzWorkUid, loggedInPersonUid)
+            }
+            view.clazzWorkPrivateComments = privateComments
+        }
+
+        newCommentItemListener.fromPerson = loggedInPersonUid
+        newCommentItemListener.entityId = clazzWorkWithSubmission.clazzWorkUid
+
+        return clazzWorkWithSubmission
+    }
+
+    override fun handleClickEdit() {
+        systemImpl.go(ClazzWorkEditView.VIEW_NAME , arguments, context)
+    }
+
+    override suspend fun onCheckEditPermission(account: UmAccount?): Boolean {
+        //TODO: this
+
+        val loggedInPersonUid = accountManager.activeAccount.personUid
+
+        val clazzMember: ClazzMember? = withTimeoutOrNull(2000) {
+            db.clazzMemberDao.findByPersonUidAndClazzUid(loggedInPersonUid,
+                    entity?.clazzWorkClazzUid?: 0L)
+        }
+        val isStudent = (clazzMember != null && clazzMember.clazzMemberRole == ClazzMember.ROLE_STUDENT)
+        return !isStudent
+
+    }
+
+    fun handleClickSubmit(){
+        val questionsWithOptionsAndResponse =
+                view.clazzWorkQuizQuestionsAndOptionsWithResponse?.getValue()?: listOf()
+        val newOptionsAndResponse = mutableListOf<ClazzWorkQuestionAndOptionWithResponse>()
+
+        val clazzWorkWithSubmission = entity
+        GlobalScope.launch {
+            for (everyResult in questionsWithOptionsAndResponse) {
+                val response = everyResult.clazzWorkQuestionResponse
+                if(response.clazzWorkQuestionResponseUid == 0L) {
+                    response.clazzWorkQuestionResponseUid =
+                            db.clazzWorkQuestionResponseDao.insertAsync(response)
+                }else{
+                    db.clazzWorkQuestionResponseDao.updateAsync(response)
+                }
+                everyResult.clazzWorkQuestionResponse = response
+                newOptionsAndResponse.add(everyResult)
+            }
+
+            val loggedInPersonUid = accountManager.activeAccount.personUid
+            val clazzMember: ClazzMember? = withTimeoutOrNull(2000){
+                db.clazzMemberDao.findByPersonUidAndClazzUid(loggedInPersonUid,
+                        entity?.clazzWorkClazzUid?:0L)
+            }
+
+            var submission = entity?.clazzWorkSubmission
```

Comment:
I think it would be better style to use the elvis operator - e.g.

```
var submission = entity?.clazzWorkSubmission ?: ClazzWorkSubmission().apply {
   clazzWorkSubmissionClazzWorkUid = clazzWorkWithSubmission?.clazzWorkUid ?: 0L
   clazzWorkSubmissionClazzMemberUid = clazzMember?.clazzMemberUid ?: 0L
   clazzWorkSubmissionDateTimeFinished = getSystemTimeInMillis()
   clazzWorkSubmissionInactive = false
   clazzWorkSubmissionPersonUid = loggedInPersonUid
}
```